File: result/metagpt/claude/humaneval/98.c
#include <stdio.h>
#include <string.h>

int count_upper(const char* s) {
    const char* uvowel = "AEIOU";
    int count = 0;
    int length = strlen(s);
    
    for (int i = 0; i * 2 < length; i++) {
        if (strchr(uvowel, s[i * 2]) != NULL) {
            count++;
        }
    }
    
    return count;
}

File: result/metagpt/deepseek/humaneval/98.rs
pub fn count_upper(s: &str) -> usize {
    const UVOWEL: &str = "AEIOU";
    s.chars()
        .enumerate()
        .filter(|(i, c)| i % 2 == 0 && UVOWEL.contains(*c))
        .count()
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_count_upper() {
        assert_eq!(count_upper("aAbBcCdDeE"), 2);
        assert_eq!(count_upper("ABCDEF"), 3);
        assert_eq!(count_upper("xyz"), 0);
    }
}